MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,920 @@ If the requested :code:`timeout_nsec` exceeds
 :code:`YDB_MAX_TIME_NSEC`, the function returns
 :code:`YDB_ERR_TIME2LONG`; otherwise it returns :code:`YDB_OK`.
 
+=================
+Programming in Go
+=================
+
+Programming YottaDB in the `Go language <https://golang.org/>`_ is
+accomplished through a wrapper that uses `cgo
+<https://golang.org/cmd/cgo/>`_ to provide the Go programmer with a
+YottaDB package. YottaDB C functions are wrapped by Go methods where a
+method can meaningfully be associated with a Go structure, and Go
+functions otherwise.
+
+As the Go language has important differences from C (for example it
+lacks macros, but has methods), below are Go-specific sections of the
+`Quick Start`_, `Concepts`_, `Symbolic Constants`_, `Data Structures &
+Type Definitions`_ and `Simple API`_ sections above. The sections
+below specific to Go are intended to supplement their general (C)
+counterparts, rather than stand-alone sections.
+
+Application programs should not directly use the YottaDB C API
+structures and functions (those prefixed by :code:`C.`), but should
+instead use the structures, methods and functions exposed by the
+YottaDB Go wrapper. :code:`C.` prefixed structures and functions are
+mentioned only for clarity in documentation and brevity of
+explanation. For example, :code:`C.ydb_buffer_t` is the C
+:code:`ydb_buffer_t` structure defined in `Data Structures & Type
+Definitions`_ and :code:`C.ydb_lock_s()` is the Simple API
+`ydb_lock_s()`_ function.
+
+All subsections of the `Programming in Go` section are prefixed with
+“Go” to ensure unique names for hyperlinking.
+
+Go Quick Start
+==============
+
+The YottaDB Go wrapper requires a minimum YottaDB version of r1.20 and
+a minimum Go version of 1.8.3 [need to verify / state what is tested
+and Supported]. The following additional Go packages are also
+required: XYZ [currently no additional packages are believed to be
+required, but should document one way or the other].
+
+The Go Quick Start assumes that YottaDB has already been installed as
+described in the `Quick Start`_ section. After completing step 1
+(Installing YottaDB), install the Go wrapper:
+
+Download the Go wrapper from XYZ [provide URL when released].  Unpack
+the contents in its own directory (e.g, $HOME/go/src/yottadb, and
+ensure that directory is in the search path for packages.
+
+Then after step 2 (Choose a directory for your default
+environment and initialize it) in the `Quick Start`_ section:
+
+3. Put your GO program in the :code:`$ydb_dir` directory, XYZ
+   (instructions to include headers).  As a sample program,
+   you can download the `wordfreq.go
+   <https://raw.githubusercontent.com/YottaDB/YottaDBtest/master/simpleapi/inref/wordfreq.c>`_
```

Comment:
This link points to the C code.

Reply:
True. The Go code doesn't exist yet because the Go wrapper doesn't exist yet. We're drafting the user documentation to specify the API.